tests/unit/hooks/test_postgresql.py

```diff
@@ -594,7 +624,233 @@ def test_restore_database_dump_runs_pg_restore_with_username_and_password():
     ).once()
 
     module.restore_database_dump(
-        database_config, 'test.yaml', {}, dry_run=False, extract_process=extract_process
+        database_config,
+        'test.yaml',
+        {},
+        dry_run=False,
+        extract_process=extract_process,
+        connection_params={
+            'hostname': None,
+            'port': None,
+            'username': None,
+            'password': None,
+        },
+    )
+
+
+def test_make_extra_environment_with_cli_password_sets_correct_password():
+    database = {'name': 'foo', 'restore_password': 'trustsome1', 'password': 'anotherpassword'}
+
+    extra = module.make_extra_environment(
+        database, restore_connection_params={'password': 'clipassword'}
+    )
+
+    assert extra['PGPASSWORD'] == 'clipassword'
```

Comment:
What about the case where there's no CLI password and it's just a `restore_connection_params` password? Worth testing too?

Reply:
restore_connection_params password is the password we get through the CLI though. Did you mean no config password?

Reply:
Yes, my bad. That's what I meant!

Reply:
Actually, looking at the function under test, I think I meant: What about the case where there's no CLI password and it's just a config password?

Reply:
Oh yeah, I remember now, I left it out because I thought `test_restore_database_dump_runs_pg_restore_with_username_and_password` tests the same thing doesn't it?

Reply:
Sorry, I'm being totally unclear. I mean the case where there's no CLI password and there's only a _restore_ config password.. (Or a restore config password + a standard `password`.)